hudi-common/src/main/java/org/apache/hudi/common/util/BaseFileUtils.java

```diff
@@ -67,6 +70,61 @@ public static BaseFileUtils getInstance(HoodieFileFormat fileFormat) {
     throw new UnsupportedOperationException(fileFormat.name() + " format not supported yet.");
   }
 
+  /**
+   * Aggregate column range statistics across files in a partition.
+   *
+   * @param fileColumnRanges List of column range statistics for each file in a partition
+   */
+  public static <T extends Comparable<T>> HoodieColumnRangeMetadata<T> getColumnRangeInPartition(@Nonnull List<HoodieColumnRangeMetadata<T>> fileColumnRanges) {
+    if (fileColumnRanges.size() == 1) {
+      // Only one parquet file, we can just return that range.
+      return fileColumnRanges.get(0);
+    }
+    // There are multiple files. Compute min(file_mins) and max(file_maxs)
+    return fileColumnRanges.stream()
+        .sequential()
+        .reduce(BaseFileUtils::mergeRanges).get();
+  }
+
+  private static <T extends Comparable<T>> HoodieColumnRangeMetadata<T> mergeRanges(HoodieColumnRangeMetadata<T> one,
```

Comment:
Do we merge column ranges between the base and log files (e.g., log file contains updates and deletes)?  Is that covered in this PR or by existing column stats logic (how does that work)?

Reply:
Got it.  I'm trying to understand how col stats are merged between the base and log files, e.g., for custom payload, because we may not be able to simply take the minimum of all minimum values or the maximum of all maximum values from the base and log files, as there can be deletes and there can be event-time based merging.  We can discuss this in a separate thread.

Reply:
We follow the same approach as for colstats - https://github.com/apache/hudi/blob/f99b181a04e2ee549009d38deca7597219c4ce46/hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java#L372
If the new record is for delete, then we ignore the previous one. If previous record is for delete, then we take the newer one, otherwise just merge the stats. I don't think we do event-time based merging for MDT.